lib/segment/Cargo.toml

```diff
@@ -60,6 +60,7 @@ quantization = { git = "https://github.com/qdrant/quantization.git" }
 validator = { version = "0.16", features = ["derive"] }
 chrono = { version = "0.4.26", features = ["serde"] }
 smol_str = "0.2.0"
+enum_dispatch = "0.3.12"
```

Comment:
Please don't use it. It breaks autocompletion and many other things in IDE.
Explicit enums definitions were used _on purpose_

Reply:
Sad that some macros seem to be a tough issue for Jetbrains. I'll stop using it then.